### PR TITLE
Fix failing file handler test

### PR DESCRIPTION
**Pull request by sruti1312**
Signed-off-by: Sruti Parthiban <partsrut@amazon.com>

**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
https://github.com/opensearch-project/performance-analyzer/issues/39
Fix failing event file handler test. Formatting changes was brought in by spotless.

**Describe the solution you are proposing**

**Describe alternatives you've considered**

**Additional context**

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/performanceanalyzer/PerformanceAnalyzerPlugin.java [hunk 1]
```diff
@@ -269,6 +269,7 @@ public PerformanceAnalyzerPlugin(final Settings settings, final java.nio.file.Pa
                         eventLogFileHandler,
                         MetricsConfiguration.SAMPLING_INTERVAL,
                         QUEUE_PURGE_INTERVAL_MS,
+                        0,
```
Comment (khushbr) [inline src/main/java/org/opensearch/performanceanalyzer/PerformanceAnalyzerPlugin.java:272]: Why are we passing the `lastCleanupTimeBucket` as an argument to `EventLogQueueProcessor` ?

Comment (sruti1312) [reply src/main/java/org/opensearch/performanceanalyzer/PerformanceAnalyzerPlugin.java]: Updated the check in the cleanup method

**Comment by jotok** [comment]
Is there an issue we can link to in the PR description?

File: src/main/java/org/opensearch/performanceanalyzer/writer/EventLogQueueProcessor.java [hunk 2]
```diff
@@ -187,16 +188,25 @@ public void purgeQueueAndPersist() {
...
+                    PerformanceAnalyzerMetrics.getTimeInterval(System.currentTimeMillis());
+            if (currCleanupTimeBucket - lastCleanupTimeBucket > filesCleanupPeriodicityMillis) {
+                // Get list of files(time buckets) for purging, considered range :
+                // [lastCleanupTimeBucket, currCleanupTimeBucket)
+                List<String> filesForCleanup =
+                        LongStream.range(lastCleanupTimeBucket, currCleanupTimeBucket)
+                                .filter(
+                                        timeMillis ->
+                                                timeMillis % MetricsConfiguration.SAMPLING_INTERVAL
+                                                        == 0)
```
Comment (jotok) [inline src/main/java/org/opensearch/performanceanalyzer/writer/EventLogQueueProcessor.java:204]: It seems a little inefficient to test every number. What if we did something like this?
```java
List<String> filesForCleanup = new ArrayList<>();

// ugly expression to handle the case when lastCleanupTimeBucket is divisible by SAMPLING_INTERVAL
// If I did this correctly, then k should satisfy the following: 
//   (k - 1) * SAMPLING_INTERVAL < lastCleanupTimeBucket <= k * SAMPLING_INTERVAL
long k = lastCleanupTimeBucket / SAMPLING_INTERVAL +
        lastCleanupTimebucket % SAMPLING_INTERVAL == 0 ? 0L : 1L;

for (int tick = k * SAMPLING_INTERVAL; tick < currCleanupTimeBucket; tick += SAMPLING_INTERVAL) {
  filesForCleanup.add(String::valueOf(tick));
}

```

**Review by jotok — APPROVED** [review]

**Review by kkhatua — APPROVED** [review]
LGTM. 
Recommendation would have been to just fetch the list of "obsolete" files from the FS, but with a 1sec iteration, this might be an overhead. Probably the better approach is to test on the FS every 5 mins. But this isn't a blocker.

**Pull request merged**